src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/MSBuildNuGetProjectProvider.cs

```diff
@@ -56,8 +56,6 @@ public MSBuildNuGetProjectProvider(IVsProjectThreadingService threadingService)
             Assumes.Present(vsProjectAdapter);
             Assumes.Present(context);
 
-            await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
No casting/UI thread affinity in this method, only the async methods called by this method have UI thread affinity.